subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c

```diff
@@ -188,12 +197,49 @@ int img_mgmt_get_next_boot_slot(int image, enum img_mgmt_next_boot_type *type)
 	return slot;
 }
 #else
+
+#if defined(CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT)
+
+static int read_directxip_state(int slot)
+{
+	struct boot_swap_state bss;
+	int fa_id = img_mgmt_flash_area_id(slot);
+	const struct flash_area *fa;
+	int rc = 0;
+
+	rc = flash_area_open(fa_id, &fa);
+	if (rc < 0) {
+		return rc;
+	}
+	rc = boot_read_swap_state(fa, &bss);
+	flash_area_close(fa);
+	if (rc != 0) {
+		LOG_ERR("Failed to read state of slot %d\n", slot);
```

Comment:
Should we map positive values to negatives here?

Reply:
Done